designs/ami-selector.md

```diff
@@ -0,0 +1,62 @@
+# Background
+
+The `AMISelector` field of the [`v1alpha1` `AWSNodeTemplate` resource](/pkg/apis/v1alpha1/awsnodetemplate.go)
+is a key-value map with some special key handling (e.g. `aws-ids` is used to pass image IDs).
+
+Most relevant to this design document was the `name` field, which was passed as the `name` filter to
+[`DescribeImages` API](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeImages.html)
+call. This field was removed as a special case in #2978, as there is a risk that without filtering
+by owner, AMI impersonation could happen with various risks (e.g. security vulnerabilities, malware
+such as mining, or cost implications).
+
+While the `Name` tag is still usable, AMI tags are private to an account (so if you want
+to use the AWS bottlerocket AMIs, you'd need to tag them yourself in each account that
+you use)
+
+# Solutions
+
+1. Restore the `name` special case and add a new `owners` special case that is passed
```

Comment:
Hey, can you change this option and update the docs based on our discussion? Just some points for refresher:
    1. Change all special casing to be `aws::` prefixed.
    2. `aws-ids` and `aws::ids` will be supported to not break users.
    3. *Caveat:* System tags are not currently tagged on AMIs used with EKS Optimized AMIs. This could be added in the future. We’ll make it `aws::` instead of `aws:` so that we can still use the special case, and be flexible to a possible system tag addition in the future.
    4. Long term we need a design doc to implement given the feature request here: https://github.com/aws/karpenter/issues/3326 (Option 2)